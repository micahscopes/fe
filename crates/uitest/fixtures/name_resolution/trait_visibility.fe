fn foo(x: i32) -> i32 {
    x.foo()
    inner::Foo::foo(x)
}

mod inner {
    trait Foo {
        fn foo(self) -> Self
    }

    impl Foo for i32 {
        fn foo(self) -> i32 {
            self
        }
    }
}
