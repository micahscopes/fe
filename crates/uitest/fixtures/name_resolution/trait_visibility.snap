---
source: crates/uitest/tests/name_resolution.rs
expression: diags
input_file: crates/uitest/fixtures/name_resolution/trait_visibility.fe
---
error[2-0003]: `Foo` is not visible
  ┌─ trait_visibility.fe:3:12
  │
3 │     inner::Foo::foo(x)
  │            ^^^ `Foo` is not visible
  ·
7 │     trait Foo {
  │           --- `Foo` is defined here
